### PR TITLE
DOCSP-20681 ssh and socks5

### PR DESCRIPTION
**Pull request by jocelyn-mendez1**
Proxy and SSH tunnel connection tab in the new Compass connection form. 

JIRA: https://jira.mongodb.org/browse/DOCSP-20681

STAGING: https://docs-mongodbcom-staging.corp.mongodb.com/compass/docsworker-xlarge/DOCSP-20681/connect/ssh-connection/

### CODE REVIEW

File: source/connect/ssh-connection.txt [hunk 1]
```diff
@@ -11,3 +11,157 @@ Proxy / SSH Tunnel Connection Tab
    :backlinks: none
    :depth: 1
    :class: singlecol
+
+The :guilabel:`Proxy / SSH Tunnel` tab allows you to connect your deployment using
```
Comment (jason-price-mongodb) [inline source/connect/ssh-connection.txt:15]: Should this be "...to connect to your..."? Ditto throughout this update for similar wording.

File: source/connect/ssh-connection.txt [hunk 2]
```diff
@@ -11,3 +11,157 @@ Proxy / SSH Tunnel Connection Tab
...
+         :header-rows: 1 
+         :widths: 50 50 
+      
+         * - Option
+           - Description 
+
+         * - SSH with Password 
+           - Connects a deployment using a SSH Tunnel and has the option to secure 
+             the connection with a password. For information on SSH connections and 
+             the fields found under this connection option, see :ref:`<ssh-fields>`.
```
Comment (jason-price-mongodb) [inline source/connect/ssh-connection.txt:44]: "fields for this connection option"

File: source/connect/ssh-connection.txt [hunk 3]
```diff
@@ -11,3 +11,157 @@ Proxy / SSH Tunnel Connection Tab
...
+
+         * - SSH with Identity File 
+           - Connects a deployment using a SSH Tunnel and uses an Identity File
```
Comment (jason-price-mongodb) [inline source/connect/ssh-connection.txt:47]: "Connects to a deployment...."?

File: source/connect/ssh-connection.txt [hunk 4]
```diff
@@ -11,3 +11,157 @@ Proxy / SSH Tunnel Connection Tab
...
+
+         * - SSH with Password 
+           - Connects a deployment using a SSH Tunnel and has the option to secure 
+             the connection with a password. For information on SSH connections and 
+             the fields found under this connection option, see :ref:`<ssh-fields>`.
+
+         * - SSH with Identity File 
+           - Connects a deployment using a SSH Tunnel and uses an Identity File 
+             to secure the connection. For information on SSH connections and 
+             the fields found under this connection option, see :ref:`<ssh-fields>`.
```
Comment (jason-price-mongodb) [inline source/connect/ssh-connection.txt:49]: "fields for this connection option"

File: source/connect/ssh-connection.txt [hunk 5]
```diff
@@ -11,3 +11,157 @@ Proxy / SSH Tunnel Connection Tab
...
+         * - SSH Hostname
+           - Enter the bastion (jumpbox) hostname. This is the unique identifier 
+             (Fully Qualified Domain Name, or FQDN) for the machine to be accessed. 
+
+         * - SSH Port
+           - Provide the port used for the SSH connection. This defaults to 22, 
+             the standard port for SSH.
+
+         * - SSH Username
+           - The username of the profile to log into on the remote system. This
```
Comment (jason-price-mongodb) [inline source/connect/ssh-connection.txt:80]: "This is the user for which ..."

File: source/connect/ssh-connection.txt [hunk 6]
```diff
@@ -11,3 +11,157 @@ Proxy / SSH Tunnel Connection Tab
...
+         * - SSH Username
+           - The username of the profile to log into on the remote system. This 
+             will be the user for which you want to establish the SSH connection.
+
+         * - SSH Password 
+           - (Optional) The password used to secure the SSH connection. This is 
+             required if you are not using an identity file.
+
+         * - SSH Identity File
+           - Select the file from which the identity (private key) for SSH public
```
Comment (jason-price-mongodb) [inline source/connect/ssh-connection.txt:88]: "The file that the identity ..."

File: source/connect/ssh-connection.txt [hunk 7]
```diff
@@ -11,3 +11,157 @@ Proxy / SSH Tunnel Connection Tab
...
+             - ``id_ed25519``
+ 
+             - ``id_rsa``
+
+             On Windows, the location of the identify files depends on your choice 
+             of SSH client, such as PuTTY. 
+
+         * - SSH Passphrase
+           - (Optional) If your private key is encrypted, provide the passphrase 
+             used to decrypt your private key (stored in the specified identity file).
```
Comment (jason-price-mongodb) [inline source/connect/ssh-connection.txt:108]: "private key, which is stored"

File: source/connect/ssh-connection.txt [hunk 8]
```diff
@@ -11,3 +11,157 @@ Proxy / SSH Tunnel Connection Tab
...
+      .. list-table::
+         :header-rows: 1 
+         :widths: 50 50
+      
+         * - Field
+           - Description
+
+         * - Proxy Hostname
+           - Specifies a domain name, IPv4, or IPv6 address on which a socks5 
+             proxy is listening
```
Comment (jason-price-mongodb) [inline source/connect/ssh-connection.txt:137]: Add period.

File: source/connect/ssh-connection.txt [hunk 9]
```diff
@@ -11,3 +11,157 @@ Proxy / SSH Tunnel Connection Tab
...
+
+         * - Proxy Tunnel Port
+           - (Optional) Specifies a TCP port number. This defaults to 1080.
```
Comment (jason-price-mongodb) [inline source/connect/ssh-connection.txt:140]: "Default is 1080."

File: source/connect/ssh-connection.txt [hunk 10]
```diff
@@ -11,3 +11,157 @@ Proxy / SSH Tunnel Connection Tab
...
+
+         * - Proxy Hostname
+           - Specifies a domain name, IPv4, or IPv6 address on which a socks5 
+             proxy is listening
+
+         * - Proxy Tunnel Port
+           - (Optional) Specifies a TCP port number. This defaults to 1080.  
+         
+         * - Proxy Username
+           - (Optional) The username used to authenticate the connection to the proxy
```
Comment (jason-price-mongodb) [inline source/connect/ssh-connection.txt:143]: "the connection with the proxy"?

File: source/connect/ssh-connection.txt [hunk 11]
```diff
@@ -11,3 +11,157 @@ Proxy / SSH Tunnel Connection Tab
...
+
+         * - Proxy Tunnel Port
+           - (Optional) Specifies a TCP port number. This defaults to 1080.  
+         
+         * - Proxy Username
+           - (Optional) The username used to authenticate the connection to the proxy 
+             server.  
+         
+         * - Proxy Password
+           - (Optional) The password used to authenticate the connection to the
```
Comment (jason-price-mongodb) [inline source/connect/ssh-connection.txt:147]: See prev comment.

File: source/connect/ssh-connection.txt [hunk 12]
```diff
@@ -11,3 +11,157 @@ Proxy / SSH Tunnel Connection Tab
...
+         * - Proxy Password
+           - (Optional) The password used to authenticate the connection to the 
+             proxy server. 
+
+      Behavior
+      ````````
+
+      - If the you specify a non-local Socks5 proxy host and provide a 
+        username and password authentication, you receive a warning notifying 
+        you that the credentials will be sent in plaintext.
```
Comment (jason-price-mongodb) [inline source/connect/ssh-connection.txt:155]: "warning stating that the credentials"

File: source/connect/ssh-connection.txt [hunk 13]
```diff
@@ -11,3 +11,157 @@ Proxy / SSH Tunnel Connection Tab
...
+
+      Behavior
+      ````````
+
+      - If the you specify a non-local Socks5 proxy host and provide a 
+        username and password authentication, you receive a warning notifying 
+        you that the credentials will be sent in plaintext. 
+
+      - If you specify a non-local Socks5 proxy host and do not enable TLS, 
+        you receive a warning notifying you that the proxy administrator will
```
Comment (jason-price-mongodb) [inline source/connect/ssh-connection.txt:158]: See prev comment.

File: source/connect/ssh-connection.txt [hunk 14]
```diff
@@ -11,3 +11,157 @@ Proxy / SSH Tunnel Connection Tab
...
+
+      .. list-table::
+         :header-rows: 1 
+         :widths: 50 50 
+      
+         * - Option
+           - Description 
+
+         * - SSH with Password 
+           - Connects a deployment using a SSH Tunnel and has the option to secure
```
Comment (jason-price-mongodb) [inline source/connect/ssh-connection.txt:42]: "Connects to a deployment"?

File: source/connect/ssh-connection.txt [hunk 15]
```diff
@@ -11,3 +11,157 @@ Proxy / SSH Tunnel Connection Tab
...
+             the fields found under this connection option, see :ref:`<ssh-fields>`.
+
+         * - SSH with Identity File 
+           - Connects a deployment using a SSH Tunnel and uses an Identity File 
+             to secure the connection. For information on SSH connections and 
+             the fields found under this connection option, see :ref:`<ssh-fields>`.
+
+         * - Socks5
+           - Connects a deployment using a seperate proxy server. For more 
+             information on the fields found under this connection option, see
```
Comment (jason-price-mongodb) [inline source/connect/ssh-connection.txt:53]: See prev comment.

File: source/connect/ssh-connection.txt [hunk 16]
```diff
@@ -11,3 +11,157 @@ Proxy / SSH Tunnel Connection Tab
...
+           - Connects a deployment using a seperate proxy server. For more 
+             information on the fields found under this connection option, see 
+             :ref:`<socks5>`. 
+
+      .. _ssh-fields:
+ 
+      SSH Connection
+      ~~~~~~~~~~~~~~
+
+      The following fields are used when connecting a deployment using
```
Comment (jason-price-mongodb) [inline source/connect/ssh-connection.txt:61]: "Use the following fields to connect to a deployment using ..."?

File: source/connect/ssh-connection.txt [hunk 17]
```diff
@@ -11,3 +11,157 @@ Proxy / SSH Tunnel Connection Tab
...
+      .. list-table::
+         :header-rows: 1 
+         :widths: 50 50
+      
+         * - Field
+           - Description
+
+         * - SSH Hostname
+           - Enter the bastion (jumpbox) hostname. This is the unique identifier 
+             (Fully Qualified Domain Name, or FQDN) for the machine to be accessed.
```
Comment (jason-price-mongodb) [inline source/connect/ssh-connection.txt:73]: "for the computer to access"

File: source/connect/ssh-connection.txt [hunk 18]
```diff
@@ -11,3 +11,157 @@ Proxy / SSH Tunnel Connection Tab
...
+      
+         * - Field
+           - Description
+
+         * - SSH Hostname
+           - Enter the bastion (jumpbox) hostname. This is the unique identifier 
+             (Fully Qualified Domain Name, or FQDN) for the machine to be accessed. 
+
+         * - SSH Port
+           - Provide the port used for the SSH connection. This defaults to 22,
```
Comment (jason-price-mongodb) [inline source/connect/ssh-connection.txt:76]: I noticed the descriptions sometimes include "The" and some omit. I would standardize on one. Please make the change throughout the update. Cheers.

File: source/connect/ssh-connection.txt [hunk 19]
```diff
@@ -11,3 +11,157 @@ Proxy / SSH Tunnel Connection Tab
...
+
+      - Users can connect to replica sets using SSH Tunnels and 
+        :ref:`TLS / SSL connections <tls-ssl-tab>`. 
+
+      .. _socks5:
+
+      Socks5
+      ~~~~~~~
+
+      The following fields are under the :guilabel:`Socks5` tab when connecting
```
Comment (jason-price-mongodb) [inline source/connect/ssh-connection.txt:125]: "The following fields are in the :guilabel:`Socks5` tab, and are used to connect to a deployment"?

File: source/connect/ssh-connection.txt [hunk 20]
```diff
@@ -11,3 +11,157 @@ Proxy / SSH Tunnel Connection Tab
...
+      - In |compass|, :guilabel:`SSH Tunnel Connections` allow users to connect 
+        to replica sets. 
+
+      - Users can connect to replica sets using SSH Tunnels and 
+        :ref:`TLS / SSL connections <tls-ssl-tab>`. 
+
+      .. _socks5:
+
+      Socks5
+      ~~~~~~~
```
Comment (jason-price-mongodb) [inline source/connect/ssh-connection.txt:123]: Remove extra ~

**Review by jason-price-mongodb — CHANGES_REQUESTED** [review]
@jocelyn-mendez1 added review comments for consideration. Cheers, Jason.

File: source/connect/ssh-connection.txt (hunk 20 above)
Comment (jocelyn-mendez1) [reply source/connect/ssh-connection.txt:123]: Done.

File: source/connect/ssh-connection.txt (hunk 19 above)
Comment (jocelyn-mendez1) [reply source/connect/ssh-connection.txt:125]: Done.

File: source/connect/ssh-connection.txt (hunk 18 above)
Comment (jocelyn-mendez1) [reply source/connect/ssh-connection.txt:76]: Done.

File: source/connect/ssh-connection.txt (hunk 17 above)
Comment (jocelyn-mendez1) [reply source/connect/ssh-connection.txt:73]: Done.

File: source/connect/ssh-connection.txt (hunk 16 above)
Comment (jocelyn-mendez1) [reply source/connect/ssh-connection.txt:61]: Done.

File: source/connect/ssh-connection.txt (hunk 15 above)
Comment (jocelyn-mendez1) [reply source/connect/ssh-connection.txt:53]: Done.

File: source/connect/ssh-connection.txt (hunk 14 above)
Comment (jocelyn-mendez1) [reply source/connect/ssh-connection.txt:42]: Done.

File: source/connect/ssh-connection.txt (hunk 12 above)
Comment (jocelyn-mendez1) [reply source/connect/ssh-connection.txt:155]: Done.

File: source/connect/ssh-connection.txt (hunk 13 above)
Comment (jocelyn-mendez1) [reply source/connect/ssh-connection.txt:158]: Done

File: source/connect/ssh-connection.txt (hunk 10 above)
Comment (jocelyn-mendez1) [reply source/connect/ssh-connection.txt:143]: Done.

File: source/connect/ssh-connection.txt (hunk 11 above)
Comment (jocelyn-mendez1) [reply source/connect/ssh-connection.txt:147]: Done.

File: source/connect/ssh-connection.txt (hunk 9 above)
Comment (jocelyn-mendez1) [reply source/connect/ssh-connection.txt:140]: Done.

File: source/connect/ssh-connection.txt (hunk 8 above)
Comment (jocelyn-mendez1) [reply source/connect/ssh-connection.txt:137]: Done.

File: source/connect/ssh-connection.txt (hunk 7 above)
Comment (jocelyn-mendez1) [reply source/connect/ssh-connection.txt:108]: Done.

File: source/connect/ssh-connection.txt (hunk 5 above)
Comment (jocelyn-mendez1) [reply source/connect/ssh-connection.txt:80]: Done.

File: source/connect/ssh-connection.txt (hunk 4 above)
Comment (jocelyn-mendez1) [reply source/connect/ssh-connection.txt:49]: Done.

File: source/connect/ssh-connection.txt (hunk 3 above)
Comment (jocelyn-mendez1) [reply source/connect/ssh-connection.txt:47]: Done.

File: source/connect/ssh-connection.txt (hunk 2 above)
Comment (jocelyn-mendez1) [reply source/connect/ssh-connection.txt:44]: Done.

File: source/connect/ssh-connection.txt (hunk 1 above)
Comment (jocelyn-mendez1) [reply source/connect/ssh-connection.txt:15]: Done.

File: source/connect/ssh-connection.txt [hunk 21]
```diff
@@ -11,3 +11,156 @@ Proxy / SSH Tunnel Connection Tab
...
+             - ``id_ed25519``
+ 
+             - ``id_rsa``
+
+             On Windows, the location of the identify files depends on your choice 
+             of SSH client, such as PuTTY. 
+
+         * - SSH Passphrase
+           - (Optional) If your private key is encrypted, provide the passphrase 
+             used to decrypt your private key, which is stored.
```
Comment (jason-price-mongodb) [inline source/connect/ssh-connection.txt:108]: "which is stored." looks odd. Stored where? I would rewrite.

File: source/connect/ssh-connection.txt [hunk 22]
```diff
@@ -11,3 +11,156 @@ Proxy / SSH Tunnel Connection Tab
...
+             server.  
+         
+         * - Proxy Password
+           - (Optional) Password used to authenticate the connection with the 
+             proxy server. 
+
+      Behavior
+      ````````
+
+      - If the you specify a non-local Socks5 proxy host and provide a
```
Comment (jason-price-mongodb) [inline source/connect/ssh-connection.txt:152]: "If the you" is incorrect.

File: source/connect/ssh-connection.txt [hunk 23]
```diff
@@ -11,3 +11,156 @@ Proxy / SSH Tunnel Connection Tab
...
+             proxy server. 
+
+      Behavior
+      ````````
+
+      - If the you specify a non-local Socks5 proxy host and provide a 
+        username and password authentication, you receive a warning 
+        stating that the credentials will be sent in plaintext. 
+
+      - If you specify a non-local Socks5 proxy host and do not enable TLS,
```
Comment (jason-price-mongodb) [inline source/connect/ssh-connection.txt:156]: "remote" not "non-local"?

File: source/connect/ssh-connection.txt [hunk 24]
```diff
@@ -11,3 +11,156 @@ Proxy / SSH Tunnel Connection Tab
...
+           - Connects to a deployment using a seperate proxy server. For more 
+             information on the fields for this connection option, see 
+             :ref:`<socks5>`. 
+
+      .. _ssh-fields:
+ 
+      SSH Connection
+      ~~~~~~~~~~~~~~
+
+      Use the following fields to connect to a deployment using
```
Comment (jason-price-mongodb) [inline source/connect/ssh-connection.txt:61]: "To connect to ...., use ..." ? Depends on how the rest of the page is written though.

File: source/connect/ssh-connection.txt [hunk 25]
```diff
@@ -11,3 +11,156 @@ Proxy / SSH Tunnel Connection Tab
...
+         * - SSH Hostname
+           - Bastion (jumpbox) hostname. This is the unique identifier 
+             (Fully Qualified Domain Name, or FQDN) for the computer to access. 
+
+         * - SSH Port
+           - Port used for the SSH connection. This defaults to 22, 
+             the standard port for SSH.
+
+         * - SSH Username
+           - User for which the profile to log into on the remote system. This
```
Comment (jason-price-mongodb) [inline source/connect/ssh-connection.txt:80]: "This is the user ..."

File: source/connect/ssh-connection.txt [hunk 26]
```diff
@@ -11,3 +11,156 @@ Proxy / SSH Tunnel Connection Tab
...
+
+         * - SSH Password 
+           - (Optional) Password used to secure the SSH connection. This is 
+             required if you are not using an identity file.
+
+         * - SSH Identity File
+           - File from which the identity (private key) for SSH public 
+             key authentication is read.
+
+             Unix or OS X: If using OpenSSH, identity files are found in the ~/.ssh
```
Comment (jason-price-mongodb) [inline source/connect/ssh-connection.txt:91]: ~/.ssh  should be in code font.

File: source/connect/ssh-connection.txt [hunk 27]
```diff
@@ -11,3 +11,156 @@ Proxy / SSH Tunnel Connection Tab
...
+
+             - ``id_dsa``
+             
+             - ``id_ecdsa``
+
+             - ``id_ed25519``
+ 
+             - ``id_rsa``
+
+             On Windows, the location of the identify files depends on your choice
```
Comment (jason-price-mongodb) [inline source/connect/ssh-connection.txt:103]: "identity" files.

File: source/connect/ssh-connection.txt [hunk 28]
```diff
@@ -11,3 +11,156 @@ Proxy / SSH Tunnel Connection Tab
...
+
+      - Users can connect to replica sets using SSH Tunnels and 
+        :ref:`TLS / SSL connections <tls-ssl-tab>`. 
+
+      .. _socks5:
+
+      Socks5
+      ~~~~~~
+
+      The following fields are in the :guilabel:`Socks5` tab, and are used
```
Comment (jason-price-mongodb) [inline source/connect/ssh-connection.txt:125]: "To connect to ..., use ..."

**Review by jason-price-mongodb — CHANGES_REQUESTED** [review]
@jocelyn-mendez1 added review comments. Cheers, Jason.

**Comment by jocelyn-mendez1** [comment]
Hi @jason-price-mongodb, thanks for the feedback! Let me know of any other possible changes. Thanks!

File: source/connect/ssh-connection.txt (hunk 28 above)
Comment (jocelyn-mendez1) [reply source/connect/ssh-connection.txt]: Done.

File: source/connect/ssh-connection.txt (hunk 27 above)
Comment (jocelyn-mendez1) [reply source/connect/ssh-connection.txt]: Done

File: source/connect/ssh-connection.txt (hunk 26 above)
Comment (jocelyn-mendez1) [reply source/connect/ssh-connection.txt]: Done.

File: source/connect/ssh-connection.txt (hunk 25 above)
Comment (jocelyn-mendez1) [reply source/connect/ssh-connection.txt]: Done.

File: source/connect/ssh-connection.txt (hunk 24 above)
Comment (jocelyn-mendez1) [reply source/connect/ssh-connection.txt]: Done

File: source/connect/ssh-connection.txt (hunk 22 above)
Comment (jocelyn-mendez1) [reply source/connect/ssh-connection.txt]: Done.

File: source/connect/ssh-connection.txt (hunk 21 above)
Comment (jocelyn-mendez1) [reply source/connect/ssh-connection.txt]: Done.

**Review by jason-price-mongodb — APPROVED** [review]
@jocelyn-mendez1 LGTM. Cheers, Jason.

**Pull request merged**